src/transformers/models/maskformer/modeling_maskformer.py

```diff
@@ -0,0 +1,1638 @@
+        # permute targets following indices
+        batch_idx = torch.cat([torch.full_like(tgt, i) for i, (_, tgt) in enumerate(indices)])
+        tgt_idx = torch.cat([tgt for (_, tgt) in indices])
+        return batch_idx, tgt_idx
+
+    def get_loss(self, loss, outputs, targets, indices, num_masks):
+        loss_map = {"labels": self.loss_labels, "masks": self.loss_masks}
+        assert loss in loss_map, f"{loss} not in loss_map"
+        return loss_map[loss](outputs, targets, indices, num_masks)
+
+    def forward(self, outputs: Dict[str, Tensor], targets: Dict[str, Tensor]) -> Dict[str, Tensor]:
+        """This performs the loss computation.
+        Parameters:
+             outputs: dict of tensors, see the output specification of the model for the format
+             targets: list of dicts, such that len(targets) == batch_size.
+                      The expected keys in each dict depends on the losses applied, see each loss' doc
+        """
+        outputs_without_aux = {k: v for k, v in outputs.items() if k != "aux_outputs"}
+
+        # Retrieve the matching between the outputs of the last layer and the targets
+        indices = self.matcher(outputs_without_aux, targets)
+
+        # Compute the average number of target masks accross all nodes, for normalization purposes
+        num_masks: Number = self.get_num_masks(targets, device=next(iter(outputs.values())).device)
+
+        # Compute all the requested losses
+        losses: Dict[str, Tensor] = {}
+        for loss in self.losses:
+            losses.update(self.get_loss(loss, outputs, targets, indices, num_masks))
+
+        # In case of auxiliary losses, we repeat this process with the output of each intermediate layer.
+        if "aux_outputs" in outputs:
+            for i, aux_outputs in enumerate(outputs["aux_outputs"]):
+                indices = self.matcher(aux_outputs, targets)
+                for loss in self.losses:
+                    l_dict = self.get_loss(loss, aux_outputs, targets, indices, num_masks)
+                    l_dict = {k + f"_{i}": v for k, v in l_dict.items()}
+                    losses.update(l_dict)
+
+        return losses
+
+    def get_num_masks(self, targets: Dict[str, Tensor], device: torch.device) -> Number:
+        # Compute the average number of target masks accross all nodes, for normalization purposes
+        num_masks: int = targets["labels"].shape[0]
+        num_masks_pt: Tensor = torch.as_tensor([num_masks], dtype=torch.float, device=device)
+        if is_dist_avail_and_initialized():
+            torch.distributed.all_reduce(num_masks_pt)
+        num_masks_clamped: Number = torch.clamp(num_masks_pt / get_world_size(), min=1).item()
+        return num_masks_clamped
+
+
+# TODO we could use our implementation of the swin transformer, that looks very similar to the authors' one,
+# with the downside of using more time to converting the weights
+# copied from original implementation
+class Mlp(nn.Module):
```

Comment:
This class should be prefixed with the model name.

Reply:
This class can be removed, along with all Swin-related classes below.

Reply:
All the swin-related code will be removed